feat: ⚙️ Refactor pattern code structure

PKief opened the pull request:
Reduce amount of code and amount of logic to apply the patterns to the file icons.

Airkro commented:
Looks like the `cosmiconfig` pattern lose loss many extensions, it's in the plan?

PKief commented:
> Looks like the `cosmiconfig` pattern lose loss many extensions, it's in the plan?

Which extensions are missing?

Airkro commented:
Run:

```ts
import { fileIcons } from './fileIcons';

const io = fileIcons.icons[220].fileNames?.join('\n');

console.log(io);
```

Get:

<table>
<tr>
<th>Before--------------------------------------</th>
<th>After-------------------------------</th>
</tr>
<tr>
<td>
<pre>
<code>.stylelintrc
.stylelintrc.json
.stylelintrc.jsonc
.stylelintrc.json5
.stylelintrc.yaml
.stylelintrc.yml
.stylelintrc.toml
.stylelintrc.js
.stylelintrc.mjs
.stylelintrc.cjs
.stylelintrc.ts
.stylelintrc.mts
.stylelintrc.cts
.config/stylelintrc
.config/stylelintrc.json
.config/stylelintrc.jsonc
.config/stylelintrc.json5
.config/stylelintrc.yaml
.config/stylelintrc.yml
.config/stylelintrc.toml
.config/stylelintrc.js
.config/stylelintrc.mjs
.config/stylelintrc.cjs
.config/stylelintrc.ts
.config/stylelintrc.mts
.config/stylelintrc.cts
stylelint.config.json
stylelint.config.jsonc
stylelint.config.json5
stylelint.config.yaml
stylelint.config.yml
stylelint.config.toml
stylelint.config.js
stylelint.config.mjs
stylelint.config.cjs
stylelint.config.ts
stylelint.config.mts
stylelint.config.cts
.stylelintignore
.stylelintcache
</code>
</pre>
</td>
<td>
<pre>
<code>.stylelintrc
.config/stylelintrc
stylelint.config
stylelint.js
stylelint.mjs
stylelint.cjs
stylelint.ts
stylelint.mts
stylelint.cts
stylelint.json
stylelint.jsonc
stylelint.json5
stylelint.yaml
stylelint.yml
stylelint.toml
.stylelintignore
.stylelintcache
</code>
</pre>
</td>
</tr>
</table>

PKief commented:
@Airkro thanks a lot, no it was not on purpose to remove some patterns. I've restored it, so that it should cover all patterns now: https://github.com/PKief/vscode-material-icon-theme/commit/fe98e7ce1e96ccd837712eb9652e1b55a748bae8. The reason for the refactoring was to make the code more self-explanatory and more explicit what kind of file names are used for each pattern. I know that the code looks a bit more comprehensive but I prefer it that way.

Airkro commented:
These lines should be removed, they are not part of Cosmiconfig.
Sorry to show them in the debug code before.

https://github.com/PKief/vscode-material-icon-theme/blob/bcc8288a059ca85f1cad5f62b037994e8c584cb6/src/icons/patterns/patterns.ts#L94-L95

Airkro commented:
> Looks like the `cosmiconfig` pattern lose loss many extensions, it's in the plan?

We should add some test cases for `function parseByPattern`, to avoid bugs like this.

PKief commented:
@Airkro thanks for the hint. I've updated the code and also added a unit test for the `parseByPattern` function :)

See commit: https://github.com/PKief/vscode-material-icon-theme/commit/d8bdd337e43b17c1b9b9ead72ef50db680283175
